databricks/koalas/series.py

```diff
@@ -4793,19 +4793,56 @@ def dot(self, other) -> Union[Scalar, "Series"]:
 
         >>> s @ s
         14
+
+        >>> kdf = ks.DataFrame({'x': [0, 1, 2, 3], 'y': [0, -1, -2, -3]})
+        >>> kdf
+           x  y
+        0  0  0
+        1  1 -1
+        2  2 -2
+        3  3 -3
+
+        >>> with ks.option_context("compute.ops_on_diff_frames", True):
+        ...     s.dot(kdf)
+        ...
+        x    14
+        y   -14
+        dtype: int64
         """
         if isinstance(other, DataFrame):
-            raise ValueError(
-                "Series.dot() is currently not supported with DataFrame since "
-                "it will cause expansive calculation as many as the number "
-                "of columns of DataFrame"
-            )
-        if self._kdf is not other._kdf:
-            if len(self.index) != len(other.index):
+            if not self.index.equals(other.index):
```

Comment:
Shall we check `same_anchor(self, other)` to avoid extra Spark jobs?

Reply:
Were you saying using `same_anchor` to replace index comparison (L4813)?

I may not fully understand. For example, `kdf` and `kser` don't have the same anchor, but they are valid inputs.
```
>>> kdf
   0  1
0  0  1
1 -2  3
2  4 -5
3  6  7
>>> kser
0    0
1    1
2    2
3    3
dtype: int64
>>> same_anchor(kser, kdf)
False
>>> with ks.option_context("compute.ops_on_diff_frames", True):
...     kser.dot(kdf)
... 
0    24
1    14
dtype: int64

```

Reply:
I mean, if `same_anchor(self, other)`, we don't need to compare `self.index.equals(other.index)` because it's always true, otherwise we should check it.

Reply:
That really makes sense. Thank you!